system/HTTP/URI.php

```diff
@@ -177,11 +177,11 @@ public static function createURIString(
                 : ltrim($path, '/');
         }
 
-        if ($query) {
+        if ($query !== null) {
```

Comment:
Tests failed.

```suggestion
        if ($query !== '' && $query !== null) {
```

There is code like:
```php
        URI::createURIString(
            $uri->getScheme(),
            $uri->getAuthority(),
            $uri->getPath(),
            $uri->getQuery(),
            $uri->getFragment()
        );
```
and `getQuery()` returns `''`.